components/gif_picker/gif_picker.jsx

```diff
@@ -59,7 +59,7 @@ export default class GifPicker extends React.PureComponent {
     }
 
     handleItemClick = (gif) => {
-        this.props.onGifClick(gif.max5mbGif);
+        this.props.onGifClick('![' + gif.title.replace(/]|\[/g, '\\$&') + '](' + gif.max5mbGif + ')');
```

Comment:
Also good call escaping the square brackets in the gif title

Reply:
It doesn't escape `\` characters a well, but that's because the flavour of markdown in MM simply looks for a `\` preceding a square bracket. So `[\\[x\]](#y)` renders like this on GH [\\[x\]](#y), but the first `[\` of that rendering is included in the link text on MM.

Markdown is a little ill-defined.

Reply:
We do plan on switching MM's Markdown to CommonMark which should match GitHub more closely, so we'll likely have to start doing that eventually, but I don't think it's a concern for now. The only way I see this becoming a concern is if Gfycat starts sending us bad data anyway.

Reply:
All user data is bad data ;)